Formula/ns-3.rb

```diff
@@ -25,43 +25,36 @@ class Ns3 < Formula
     depends_on "gcc"
   end
 
-  # `gcc version 5.4.0 older than minimum supported version 7.0.0`
+  # `g++ version 8 is now the minimum g++ compiler version supported.`
   fails_with gcc: "5"
+  fails_with gcc: "6"
+  fails_with gcc: "7"
 
   resource "pybindgen" do
     url "https://files.pythonhosted.org/packages/e0/8e/de441f26282eb869ac987c9a291af7e3773d93ffdb8e4add664b392ea439/PyBindGen-0.22.1.tar.gz"
     sha256 "8c7f22391a49a84518f5a2ad06e3a5b1e839d10e34da7631519c8a28fcba3764"
   end
 
-  # Fix ../src/lte/model/fdmt-ff-mac-scheduler.cc:1044:16: error:
-  # variable 'bytesTxed' set but not used [-Werror,-Wunused-but-set-variable]
-  # TODO: remove in the next release.
-  patch do
-    url "https://gitlab.com/nsnam/ns-3-dev/-/commit/dbd49741fcd5938edec17eddcde251b5dee25d05.diff"
-    sha256 "28e92297cfe058cfb587527dc3cfdb8ac66b51aba672be29539b6544591e2f1e"
-  end
-
   def install
     resource("pybindgen").stage buildpath/"pybindgen"
 
-    system "./waf", "configure", "--prefix=#{prefix}",
-                                 "--build-profile=release",
-                                 "--disable-gtk",
-                                 "--with-python=#{which("python3")}",
-                                 "--with-pybindgen=#{buildpath}/pybindgen"
-    system "./waf", "--jobs=#{ENV.make_jobs}"
-    system "./waf", "install"
+    system "cmake", "-S", ".", "-B", "build", *std_cmake_args, "-DCMAKE_INSTALL_RPATH=#{rpath}"
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
 
     pkgshare.install "examples/tutorial/first.cc", "examples/tutorial/first.py"
   end
 
   test do
-    system ENV.cxx, pkgshare/"first.cc", "-I#{include}/ns#{version}", "-L#{lib}",
+    system ENV.cxx, "-std=c++17", pkgshare/"first.cc",
+           "-I#{include}/ns#{version}", "-L#{lib}",
            "-lns#{version}-core", "-lns#{version}-network", "-lns#{version}-internet",
            "-lns#{version}-point-to-point", "-lns#{version}-applications",
-           "-std=c++11", "-o", "test"
+           "-o", "test"
     system "./test"
 
-    system Formula["python@3.10"].opt_bin/"python3", pkgshare/"first.py"
+    # As of ns-3.36, the python bindings do not work with python 3.9 or python 3.10
+    # https://gitlab.com/nsnam/ns-3-dev/-/blob/master/doc/manual/source/python.rst#L46-56
+    # system Formula["python@3.10"].opt_bin/"python3", pkgshare/"first.py"
```

Comment:
From that file:

> Compiling the existing bindings should work through Python 3.10.

Only:

> The API scanning framework (in particular, pygccxml) requires Python 3.6 through 3.8 but is not compatible with Python 3.9 or 3.10.

Reply:
raise [a question](https://gitlab.com/nsnam/ns-3-dev/-/issues/643) about this.

Reply:
To clarify... we did test the existing bindings with Python up to 3.10, and they should work.

If you need to re-generate the bindings you'll need Python 3.8, but regenerating the bindings is only required if you modify the ns-3 sources in `contrib` or `src`.

The problem (as pointed out in our tracker) is that you need to add `--enable-python-bindings` to the configuration.

As a side note, I'd avoid to use `cmake` directly, there's a nice build and configuration script that handles everything. It's named `ns3`.